assert/assertions.go

```diff
@@ -48,6 +48,40 @@ type ErrorAssertionFunc func(TestingT, error, ...interface{}) bool
 // Comparison is a custom function that returns true on success and false on failure
 type Comparison func() (success bool)
 
+// ErrorIsFor returns an [ErrorAssertionFunc] which tests if the error wraps target.
+func ErrorIsFor(target error) ErrorAssertionFunc {
+	return func(t TestingT, err error, msgAndArgs ...interface{}) bool {
+		if h, ok := t.(tHelper); ok {
+			h.Helper()
+		}
+
+		return ErrorIs(t, err, target, msgAndArgs...)
+	}
+}
+
+// ErrorAsFor returns an [ErrorAssertionFunc] which tests if the any error in err's tree matches target and if so, assigns it to target.
+// The returned function panics if target is not a non-nil pointer to either a type that implements error, or to any interface type.
+func ErrorAsFor(target interface{}) ErrorAssertionFunc {
+	return func(t TestingT, err error, msgAndArgs ...interface{}) bool {
+		if h, ok := t.(tHelper); ok {
+			h.Helper()
+		}
+
+		return ErrorAs(t, err, target, msgAndArgs...)
+	}
+}
+
+// ErrorTypeFor returns an [ErrorAssertionFunc] which tests whether the type of the error matches the targets type.
+func ErrorTypeFor(target interface{}) ErrorAssertionFunc {
```

Comment:
`target` should be of type `error`, not any.

```suggestion
func ErrorTypeFor(target error) ErrorAssertionFunc {
```